src/backend/src/utils/projects.utils.ts

```diff
@@ -60,3 +61,41 @@ export const getUserFullName = async (userId: number | null): Promise<string | n
   if (!user) throw new NotFoundException('User', userId);
   return `${user.firstName} ${user.lastName}`;
 };
+
+export const createChanges = async <T>(
+  wbsElementId: number,
+  originalProject: Project,
+  crId: number,
+  implementerId: number,
+  changes: { nameOfField: string; oldValue: string | number | null; newValue: string | number | null }[],
+  listChanges: { nameOfField: string; oldArray: ChangeListValue<T>[]; newArray: ChangeListValue<T>[] }[]
+) => {
+  let changesJson: ChangeCreateArgs[] = [];
+  for (const value of changes) {
+    const { nameOfField, oldValue, newValue } = value;
+    const changeJson = createChange(nameOfField, oldValue, newValue, crId, implementerId, wbsElementId);
+    if (changeJson !== undefined) {
+      changesJson.push(changeJson);
+    }
+  }
+
+  const listChangesJson: ReturnType<typeof createListChanges>[] = [];
+  for await (const value of listChanges) {
+    const { nameOfField, oldArray, newArray } = value;
+    const listChangeJson = createListChanges(nameOfField, oldArray, newArray, crId, implementerId, wbsElementId);
+
+    const capitalizedFieldName = nameOfField.charAt(0).toUpperCase() + nameOfField.slice(1); // capitalize first letter
+    const descriptionBulletIdField = `projectId${capitalizedFieldName}`;
+    await addDescriptionBullets(
+      listChangeJson.addedElements.map((descriptionBullet) => descriptionBullet.detail),
+      originalProject.id,
+      descriptionBulletIdField
+    );
+  }
+
+  listChangesJson.forEach((listChangeJson) => {
+    changesJson = changesJson.concat(listChangeJson.changes);
+  });
+
+  //
```

Comment:
@RChandler234 @Peyton-McKee What do you think of a function like is given a list of changes/list changes, and it loops through them. All the function calls to create a changeJson seem very similar